### PR TITLE
Cuda 12.5 Tests

### PR DESCRIPTION
**Pull request by johnnynunez**
cuda 12.5 is out

### CODE REVIEW

File: .pfnci/matrix.yaml [hunk 1]
```diff
@@ -364,6 +364,37 @@
...
+- project: "cupy.linux.cuda125"
+  target: "cuda125"
+  tags: ["@push", "full", "mini", "cuda125"]
+  system: "linux"
+  os: "ubuntu:20.04"
+  cuda: "12.5"
+  rocm: null
+  nccl: "2.20"
+  cutensor: "2.0"
+  cusparselt: "0.2.0"
```
Comment (kmaehashi) [inline .pfnci/matrix.yaml:378]:
```suggestion
  cusparselt: "0.6.1"
```

File: .pfnci/schema.yaml [hunk 2]
```diff
@@ -150,6 +153,7 @@ nccl:
       "12.2":  # Note: DEB package unavailable on Ubuntu 22.04
       "12.3":
       "12.4":
+      "12.5":
```
Comment (kmaehashi) [inline .pfnci/schema.yaml:156]: CUDA 12.5 requires NCCL 2.21.

File: .pfnci/schema.yaml [hunk 3]
```diff
@@ -180,11 +185,19 @@ cusparselt:
   "0.6.1":
     spec: "0.6.1.*"
     cuda:
+      "11.2":
+      "11.3":
+      "11.4":
+      "11.5":
+      "11.6":
+      "11.7":
+      "11.8":
```
Comment (kmaehashi) [inline .pfnci/schema.yaml:194]: https://docs.nvidia.com/cuda/cusparselt/release_notes.html#cusparselt-v0-5-0
> cuSPARSELt requires CUDA 12.0 or above

**Review by kmaehashi — CHANGES_REQUESTED** [review]
Thanks!

**Comment by johnnynunez** [comment]
> Also please remove this line: https://github.com/cupy/cupy/pull/8337/files#diff-c5351c21fb860332e41aa81e1599475f79f63e2bc47e0f32922c721c2cbbc749R337

what lines?
image

File: .pfnci/schema.yaml [hunk 4]
```diff
@@ -149,7 +152,11 @@ nccl:
       "12.0":  # Note: DEB package unavailable on Ubuntu 22.04
       "12.2":  # Note: DEB package unavailable on Ubuntu 22.04
       "12.3":
-      "12.4":
+      "1.24":
```
Comment (kmaehashi) [inline .pfnci/schema.yaml:155]:
```suggestion
      "12.4":
```

File: .pfnci/matrix.yaml [hunk 5]
```diff
@@ -364,6 +363,37 @@
...
+# CUDA 12.5 | Linux
+# The latest CUDA version matrix is intended to cover the highest supported combination.
+- project: "cupy.linux.cuda125"
+  target: "cuda125"
+  tags: ["@push", "full", "mini", "cuda125"]
+  system: "linux"
+  os: "ubuntu:20.04"
+  cuda: "12.5"
+  rocm: null
+  nccl: "2.20"
```
Comment (kmaehashi) [inline .pfnci/matrix.yaml:375]:
```suggestion
  nccl: "2.21"
```

**Comment by kmaehashi** [comment]
Hi, sorry for keeping you waiting! I've kicked new tests so let's see how it goes :eyes:

https://ci.preferred.jp/cupy.linux.cuda125/164614/
https://ci.preferred.jp/cupy.linux.cuda125.multi/164615/
https://ci.preferred.jp/cupy.win.cuda125/164613/

**Comment by kmaehashi** [comment]
All tests green (and without timeout), merging. Thanks again @johnnynunez!

**Comment by kmaehashi** [comment]
Will backport after #8422

**Pull request merged**